src/name-resolution.md

```diff
@@ -1,6 +1,23 @@
 # Name resolution
 
-The name resolution is a two-phase process. In the first phase, which runs
+## Basics
+
+In our programs we can refer to variables, types, functions, etc, by giving them
+a name. These names are not always unique. For example, take this valid Rust
+program:
+
+```rust
+type x = u32;
+let x: x = 1;
+let y: x = 2;
+```
+
+How do we know on line 3 whether `x` is a type (u32) or a value (1)?
+These conflicts are resolved in `librustc_resolve`.
```

Comment:
Now, the below part (L35-L36) seems redundant:

```
The name resolution lives in the librustc_resolve crate, with the meat in lib.rs and some helpers or symbol-type specific logic in the other modules.
```

I think it can be squashed with this line?

Reply:
Oh, good catch. https://github.com/rust-lang/rustc-guide/pull/590/commits/ee9a681de2e3e3bc595b87e3772d9d631c5d5e78 changes `resolved in 'librustc_resolve'` to `resolved during name resolution` (and reformats the paragraph).